src/utils/url.js

```diff
@@ -42,11 +51,26 @@ export function urlParams (url) {
   }
   queryParams.split('&').forEach(function (query) {
     if (query) {
-      query = ((query = query.split('=')) && query.length === 2 ? query : [query[0], 'true']).map(_decode)
+      query = ((query = query.split('=')) && query.length === 2 ? query : [query[0], 'true'])
       if (query[0].slice(-2) === '[]') obj[query[0] = query[0].slice(0, -2)] = obj[query[0]] || []
-      if (!obj[query[0]]) return (obj[query[0]] = _convert(query[1]))
-      isArray(obj[query[0]]) ? obj[query[0]].push(_convert(query[1])) : (obj[query[0]] = [obj[query[0]], _convert(query[1])])
+      if (!obj[query[0]]) return (obj[query[0]] = query[1])
+      isArray(obj[query[0]]) ? obj[query[0]].push(query[1]) : (obj[query[0]] = [obj[query[0]], query[1]])
     }
   })
   return obj
 }
+
+export function decodeValue (v) {
+  return v.replace(/(%[\dA-F]{2})+/gi, decodeURIComponent)
+}
+
+export function urlParams (url) {
+  const params = _allParams(url)
+  Object.keys(params).forEach((k) => _parseParam(params, k))
+  return params
+}
+
+export function getQueryParameter (url, name) {
+  const params = _allParams(url)
+  return _parseParam(params, name)
+}
```

Comment:
The treatment of the 'undefined' string seems to be odd. For example 
```
 getQueryParameter('https://www.bluenile.com/hk/zh/diamonds?name=undefined', 'name')
```

will give you `null` as value of `name` instead of 'undefined'.

Reply:
it was because of the converter there https://github.com/LiveIntent/live-connect/blob/master/src/utils/url.js#L23

Reply:
Yes, but shouldn't we get the value instead? It is a corner case, but still seems weird.

Reply:
we could change the converter to return undefined in that case to forward the exact same value

Reply:
done

Reply:
Hmm, it looks like ti was a desired behavior: https://github.com/LiveIntent/live-connect/blob/master/test/unit/utils/url.spec.js#L29. Revert?

Reply:
reverted and now it will return undefined only if the parameter is not defined